support-frontend/assets/components/contributionsReminder/contributionsReminder.jsx

```diff
@@ -60,50 +76,59 @@ class ContributionsReminder extends Component<PropTypes, StateTypes> {
   renderButtonCopy = (buttonState: ButtonState): string => {
     switch (buttonState) {
       case 'initial':
-        return 'Remind me in March';
+        return 'Set my reminder';
       case 'pending':
         return 'Pending...';
       case 'success':
         return 'Signed up';
       case 'fail':
         return 'Sorry, something went wrong';
       default:
-        return 'Remind me in March';
+        return 'Set my reminder';
     }
   }
 
   render() {
     const { email, csrf } = this.props;
+
     const requestParams = {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
         'Csrf-Token': csrf.token || '',
       },
-      body: {
-        email,
-        reminderDate: '2020-03-19 00:00:00',
-      },
+      body: JSON.stringify({
+          'email': email,
+          'reminderDate': this.state.selectedDate,
+        }
+      ),
     };
 
     if (email && csrf.token) {
-      const isSuccess = this.state.buttonState === 'success';
       const isClicked = this.state.buttonState !== 'initial';
 
       return (
-        <section className="contribution-thank-you-block">
+        <section className="contribution-thank-you-block contribution-thank-you-block--reminders">
           <h3 className="contribution-thank-you-block__title">
-            Set up a reminder for the end of the year
+            Set up a reminder to contribute again
           </h3>
           <p className="contribution-thank-you-block__message">
             {'Lots of readers choose to make single contributions at various points in the year. Opt in to receive a reminder in case you would like to support our journalism again. This will be a single email, with no obligation.'}
           </p>
 
+          <FormLabel label="Remind me in:" htmlFor={null}>
+            <Fieldset legend="Choose one of the following dates to receive your reminder:">
+              <RadioInput id="march2020" text="March 2020" name="reminder" onChange={evt => this.setDateState(evt, '2020-03-19 00:00:00', 'march2020')} />
+              <RadioInput id="june2020" text="June 2020" name="reminder" onChange={evt => this.setDateState(evt, '2020-06-01 00:00:00', 'june2020')} />
```

Comment:
Just wondering if it would be more efficient/DRYer to use a loop or map to create the radio inputs from an array of objects?

Reply:
done